### DISCUSSION TITLE
Multiselect Feature Request

### DISCUSSION THREAD

**Discussion by dustinlebsock** [iTwin/iTwinUI-react#586] [Ideas]
Hi all,

When using the iTwin-React library, I've been finding myself constantly needing a 'multiselect' ui component. Could be added as a core feature?

I've just been combining the<Checkbox/> component by passing it into the itemRenderer, but just having a single multiselect would be a lot nicer imo.

```
<Select<option>
        options={getOptions()}
        itemRenderer={(option, itemProps) => (
          <Checkbox
            label={option.label}
            checked={option.value.selected}
            onClick={() => onClick(option.value)} />
        )}
/>
```

![image](https://user-images.githubusercontent.com/25542658/158186849-c38d42b9-ba74-450b-b959-057b5641e549.png)

**Comment by veekeys** [comment]
Hello @dustinlebsock,
Thank you for using iTwinUI!

Yes, we are aware of this need and have this #344 issue  created to follow it. We just start the development of it (design first and then react implementation), so it might take some time. Currently, I cannot give any time estimates, but it is one of our top priorities.

**Reply by S-Bra** [reply]
We have the same need and added "extra content" as supported by the DropdownMenu:
- "toggle all" option when more than 3 options exist
- "search" to filter down longer lists
-  "extra content" in a special use case to define how multiple options are combined

Label based on the selected options/nothing selected
![image](https://user-images.githubusercontent.com/45755960/179678654-cad4db6f-95d3-4f06-841b-71f3bbb2ab40.png)
![image](https://user-images.githubusercontent.com/45755960/179678525-498e8e90-1f0b-4e54-a277-8eb134e046b0.png)

Search field displayed when length of options above some threshold
![image](https://user-images.githubusercontent.com/45755960/179679011-179a26d6-0f8f-4487-8387-d05a8f8f9c98.png)
